src/PlaywrightSharp.Chromium/ChromiumBrowserType.cs

```diff
@@ -25,9 +38,49 @@ public Task<IBrowser> ConnectAsync(ConnectOptions options = null)
         }
 
         /// <inheritdoc cref="IBrowserType"/>
-        public IBrowserFetcher CreateBrowserFetcher(BrowserFetcherOptions options)
+        public IBrowserFetcher CreateBrowserFetcher(BrowserFetcherOptions options = null)
         {
-            throw new NotImplementedException();
+            var downloadUrls = new Dictionary<Platform, string>
+            {
+                [Platform.Linux] = "{0}/chromium-browser-snapshots/Linux_x64/{1}/{2}.zip",
+                [Platform.MacOS] = "{0}/chromium-browser-snapshots/Mac/{1}/{2}.zip",
+                [Platform.Win32] = "{0}/chromium-browser-snapshots/Win/{1}/{2}.zip",
+                [Platform.Win64] = "{0}/chromium-browser-snapshots/Win_x64/{1}/{2}.zip",
+            };
+
+            var path = options?.Path ?? Path.Combine(Directory.GetCurrentDirectory(), ".local-chromium");
+            string host = options?.Host ?? "https://storage.googleapis.com";
+            var platform = options.Platform ?? GetPlatform();
+
+            Func<Platform, string, BrowserFetcherConfig> paramsGetter = (platform, revision) =>
+            {
+                var archiveName = string.Empty;
+                var executablePath = string.Empty;
+                if (platform == Platform.Linux)

```

Comment:
could we use here the new C# 8 switch expression?

Reply:
Converted to a switch, but I have many lines to turn them into expressions.